server/container_create.go

```diff
@@ -165,7 +166,22 @@ func addImageVolumes(ctx context.Context, rootfs string, s *Server, containerInf
 		}
 		switch s.config.ImageVolumes {
 		case config.ImageVolumesMkdir:
-			if err1 := os.MkdirAll(fp, 0755); err1 != nil {
+			imageUser := ""
```

Comment:
Let's not repeat the logic here for getting the container user. You can get it back from the spec after uid, gid are added to it and move the code around to make sure that we setup User before we add the image volumes.